c/tskit/genotypes.h

```diff
@@ -34,46 +34,150 @@ extern "C" {
 
 #define TSK_ISOLATED_NOT_MISSING (1 << 1)
 
+/**
+@brief A variant at a specfic site.
+
+@rst
+Used to generate the genotypes for a given set of samples at a given
+site.
+@endrst
+*/
 typedef struct {
+    /** @brief Unowned reference to the tree sequence of the variant */
```

Comment:
I can see the arguments for hiding it. My main motivation for documenting it is that I can imagine cases where a user wants to know something else whilst at a site and will have to have their own tree, when they could have just used this one. But this may be premature optimisation, removed in https://github.com/tskit-dev/tskit/pull/2186/commits/f52ec777a6314cc6656b4a52642ebc923132d29c